From 1adc2d2946c7a8c55a09c7a5e2e45f941fc23eec Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Tue, 30 Sep 2014 13:15:19 +0800
Subject: [PATCH] ARM: at91: mmc portA support is only for at91sam9g20ek_2mmc
 board

The MMC support will enable MCI port A. Which is only exist for 2mmc
board. So move this for 2mmc board.

Since MMC has confliction with Dataflash in the CLK pin, so this patch
will make dataflash works for at91sam9260ek board.

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 boards.cfg                      | 2 +-
 include/configs/at91sam9260ek.h | 6 ++++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/boards.cfg b/boards.cfg
index 6d0778a..57c1e5d 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -90,7 +90,7 @@ Active  arm         arm926ejs      at91        atmel           at91sam9260ek
 Active  arm         arm926ejs      at91        atmel           at91sam9260ek       at91sam9g20ek_2mmc_nandflash          at91sam9260ek:AT91SAM9G20,AT91SAM9G20EK_2MMC,SYS_USE_NANDFLASH                                                                    Stelian Pop <stelian@popies.net>
 Active  arm         arm926ejs      at91        atmel           at91sam9260ek       at91sam9g20ek_dataflash_cs0           at91sam9260ek:AT91SAM9G20,SYS_USE_DATAFLASH_CS0                                                                                   Stelian Pop <stelian@popies.net>
 Active  arm         arm926ejs      at91        atmel           at91sam9260ek       at91sam9g20ek_dataflash_cs1           at91sam9260ek:AT91SAM9G20,SYS_USE_DATAFLASH_CS1                                                                                   Stelian Pop <stelian@popies.net>
-Active  arm         arm926ejs      at91        atmel           at91sam9260ek       at91sam9g20ek_mmc                     at91sam9260ek:AT91SAM9G20,SYS_USE_MMC                                                                                             Stelian Pop <stelian@popies.net>
+Active  arm         arm926ejs      at91        atmel           at91sam9260ek       at91sam9g20ek_2mmc                    at91sam9260ek:AT91SAM9G20,AT91SAM9G20EK_2MMC,SYS_USE_MMC                                                                          Stelian Pop <stelian@popies.net>
 Active  arm         arm926ejs      at91        atmel           at91sam9260ek       at91sam9g20ek_nandflash               at91sam9260ek:AT91SAM9G20,SYS_USE_NANDFLASH                                                                                       Stelian Pop <stelian@popies.net>
 Active  arm         arm926ejs      at91        atmel           at91sam9260ek       at91sam9xeek_dataflash_cs0            at91sam9260ek:AT91SAM9XE,SYS_USE_DATAFLASH_CS0                                                                                    Stelian Pop <stelian@popies.net>
 Active  arm         arm926ejs      at91        atmel           at91sam9260ek       at91sam9xeek_dataflash_cs1            at91sam9260ek:AT91SAM9XE,SYS_USE_DATAFLASH_CS1                                                                                    Stelian Pop <stelian@popies.net>
diff --git a/include/configs/at91sam9260ek.h b/include/configs/at91sam9260ek.h
index 0f6ef31..9193312 100644
--- a/include/configs/at91sam9260ek.h
+++ b/include/configs/at91sam9260ek.h
@@ -90,7 +90,6 @@
 #define CONFIG_CMD_PING		1
 #define CONFIG_CMD_DHCP		1
 #define CONFIG_CMD_NAND		1
-#define CONFIG_CMD_MMC
 #define CONFIG_CMD_FAT
 #define CONFIG_CMD_USB		1
 
@@ -133,8 +132,8 @@
 # define CONFIG_MACH_TYPE MACH_TYPE_AT91SAM9260EK
 #endif
 
-/* DataFlash */
 #ifndef CONFIG_AT91SAM9G20EK_2MMC
+/* DataFlash */
 #define CONFIG_ATMEL_DATAFLASH_SPI
 #define CONFIG_HAS_DATAFLASH		1
 #define CONFIG_SYS_SPI_WRITE_TOUT		(5*CONFIG_SYS_HZ)
@@ -142,6 +141,9 @@
 #define CONFIG_SYS_DATAFLASH_LOGIC_ADDR_CS0	0xC0000000	/* CS0 */
 #define CONFIG_SYS_DATAFLASH_LOGIC_ADDR_CS1	0xD0000000	/* CS1 */
 #define AT91_SPI_CLK			15000000
+#else
+/* Enable MMC. The MCCK is conflicted with DataFlash */
+#define CONFIG_CMD_MMC
 #endif
 
 #ifdef CONFIG_AT91SAM9G20EK
-- 
1.9.1

